### PR TITLE
Issue #16410: Validate that the userFilter and groupFilter contain an…

### LINKED ISSUE #16410: Improve messaging in ldapRegistry-3.0 when userFilter and groupFilter do not contain an AVA with %v
The ldapRegistry-3.0 configuration attributes `userFilter` and `groupFilter` are invalid when they do not contain an Attribute Value Assertion (AVA) with a `%v`. The `%v` AVA is required so that LDAP registry can replace the `%v` with the name of the user or group being searched for.

For example, the following filters are valid:

`userFilter="(&(uid=%v)(objectclass=inetorgperson))"`
`groupFilter="(&(cn=%v)(objectclass=groupofnames))"`

The following filters are invalid, because they do not contain the %v in the AVA:

`userFilter="(&(uid=randy)(objectclass=inetorgperson))"`
`groupFilter="(&(cn=admins)(objectclass=groupofnames))"`

### PR DESCRIPTION
**Pull request by jvanhill**
… AVA with %v.

fixes #16410

### CODE REVIEW

**Comment by jvanhill** [comment]
#build

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/com/ibm/ws/security/wim/adapter/ldap/resources/LdapUtilMessages.nlsprops [hunk 1]
```diff
@@ -279,3 +279,7 @@ IDMAP_INVALID_FORMAT.useraction=Update the userIdMap value so that it adheres to
 KRB5_SERVICE_NOT_AVAILABLE=CWIML4521E: The {0} LDAP registry attempted to bind to the LDAP server by using Kerberos credentials for the {1} principal name, but the Kerberos service that provides these credentials is not available. The bind authentication mechanism is {2}.
 KRB5_SERVICE_NOT_AVAILABLE.explanation=The Kerberos service was not available when the LDAP registry attempted to bind to the LDAP server. If the Kerberos service is not available, no users can authenticate.
 KRB5_SERVICE_NOT_AVAILABLE.useraction=Review the LDAP registry configuration settings for the bindAuthMechanism, krb5Principal, and krb5TicketCache attributes as well as the Kerberos configuration. Restart the server.
+
+FILTER_MISSING_PERCENT_V=CWIML4523E: The {0} value for {1} is invalid. It requires an attribute value assertion where the value assertion is =%v. For example, {2}.
+FILTER_MISSING_PERCENT_V.explanation=The LDAP registry will replace the %v with the value to search for when doing searches for users or groups.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.security.wim.adapter.ldap/resources/com/ibm/ws/security/wim/adapter/ldap/resources/LdapUtilMessages.nlsprops:284]: Change,
"The LDAP registry will replace the %v with the value to search for when doing searches for users or groups."
to,
"The LDAP registry replaces the %v with the value to search for when you search for users or groups."

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -114,10 +114,10 @@ failoverServer.port.desc=LDAP failover server port.
 #==== LDAP FILTER CONFIGURATION =====
 
 userFilter=User filter
-userFilter.desc=An LDAP filter clause for searching the user registry for users.
+userFilter.desc=An LDAP filter clause for searching the user registry for users. When defined, this filter requires an Attribute Value Assertion (AVA) containing a %v. For example, uid=%v. The %v in the AVA will be replaced during searches with the user or user pattern being searched for.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties:117]: Change,
"The %v in the AVA will be replaced during searches with the user or user pattern being searched for."
to,
"During searches, the %v in the AVA is replaced with the user or user pattern being searched for."

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -114,10 +114,10 @@ failoverServer.port.desc=LDAP failover server port.
 #==== LDAP FILTER CONFIGURATION =====
 
 userFilter=User filter
-userFilter.desc=An LDAP filter clause for searching the user registry for users.
+userFilter.desc=An LDAP filter clause for searching the user registry for users. When defined, this filter requires an Attribute Value Assertion (AVA) containing a %v. For example, uid=%v. The %v in the AVA will be replaced during searches with the user or user pattern being searched for.  
 
 groupFilter=Group filter
-groupFilter.desc=An LDAP filter clause for searching the user registry for groups.
+groupFilter.desc=An LDAP filter clause for searching the user registry for groups.  When defined, this filter requires an Attribute Value Assertion (AVA) containing a %v. For example, cn=%v. The %v in the AVA will be replaced during searches with the group or group pattern being searched for.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties:120]: Change,
"The %v in the AVA will be replaced during searches with the user or user pattern being searched for."
to,
"During searches, the %v in the AVA is replaced with the user or user pattern being searched for."

**Review by ManasiGandhi — CHANGES_REQUESTED** [review]
Minor edit suggestions.

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/com/ibm/ws/security/wim/adapter/ldap/resources/LdapUtilMessages.nlsprops (hunk 1 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.wim.adapter.ldap/resources/com/ibm/ws/security/wim/adapter/ldap/resources/LdapUtilMessages.nlsprops]: Made the changes.

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties]: Made the changes.

**Comment by jvanhill** [comment]
#libby
#build

**Comment by jvanhill** [comment]
Previous build had not started prior to me making the latest push.

**Review by jacksonleonard — APPROVED** [review]
Looks great, very clear messages and description text. Thanks Jesse!

**Review by ManasiGandhi — APPROVED** [review]

File: dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPRegressionTest.java [hunk 4]
```diff
@@ -396,6 +396,27 @@ public void testBetaFenceForBindAuthMech() throws Exception {
         Log.info(c, methodName, "Finished Liberty server update");
 
         assertFalse("Did not find CWWKE0701E for beta fence in log", libertyServer.findStringsInLogs("CWWKE0701E").isEmpty());
+    }
+
+    @Test
```
Comment (kristip17) [inline dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPRegressionTest.java:401]: Javadoc or comments for these new tests?

**Review by kristip17 — APPROVED** [review]

File: dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPRegressionTest.java (hunk 4 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LDAPRegressionTest.java:404]: Added comments.

**Comment by jvanhill** [comment]
No need to rerun as the latest push only included changes to comments in the unit test.

**Comment by jvanhill** [comment]
#libby

**Pull request merged**